pkg/controller/replicaset/replica_set.go

```diff
@@ -650,3 +608,81 @@ func (rsc *ReplicaSetController) syncReplicaSet(key string) error {
 	}
 	return manageReplicasErr
 }
+
+func (rsc *ReplicaSetController) claimPods(rs *extensions.ReplicaSet, filteredPods []*v1.Pod) ([]*v1.Pod, error) {
+	selector, err := metav1.LabelSelectorAsSelector(rs.Spec.Selector)
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("Error converting pod selector to selector: %v", err))
```

Comment:
In the original code, we would abort syncReplicaSet but report success to prevent retries, since this problem is unlikely to resolve itself. To preserve that behavior, I think we'll need to move the selector conversion back into syncReplicaSet. Sorry for not noticing earlier.

Reply:
Yup, i think moving back the selector conversion to `syncReplicaSet` is better. Not sure how to handle panic within `rsc.claimPods()`.